### PR TITLE
chore: add scaling functions to fpdecimal

### PR DESCRIPTION
**Pull request by Tofel**

### CODE REVIEW

File: packages/injective-math/src/fp_decimal/from_str.rs [hunk 1]
```diff
@@ -14,7 +14,7 @@ impl FromStr for FPDecimal {
     /// This never performs any kind of rounding.
     /// More than 18 fractional digits, even zeros, result in an error.
     fn from_str(input: &str) -> Result<Self, Self::Err> {
-        let sign = if input.starts_with('-') { 0 } else { 1 };
+        let sign = i8::from(!input.starts_with('-'));
```
Comment (gorgos) [inline packages/injective-math/src/fp_decimal/from_str.rs]: trying this for the second time now, the old code is more readable, I know you get this suggested from somewhere, I don't though, maybe need to fix your IDE?

**Review by gorgos — CHANGES_REQUESTED** [review]
LGTM just one comment

File: packages/injective-math/src/fp_decimal/from_str.rs (hunk 1 above)
Comment (Tofel) [reply packages/injective-math/src/fp_decimal/from_str.rs]: This is coming from `clippy`. What version are you using? I have `clippy 0.1.66 (69f9c33d 2022-12-12)`

Comment (gorgos) [reply packages/injective-math/src/fp_decimal/from_str.rs]: `clippy 0.1.69 (84c898d6 2023-04-16)`

**Review by gorgos — APPROVED** [review]

**Pull request merged**